packages/dashboard-core-plugins/src/panels/ConsolePanel.test.tsx

```diff
@@ -2,49 +2,98 @@ import React from 'react';
 import { render } from '@testing-library/react';
 import { CommandHistoryStorage } from '@deephaven/console';
 import type { Container } from '@deephaven/golden-layout';
+import { IdeConnection, IdeSession } from '@deephaven/jsapi-types';
 import { ConsolePanel } from './ConsolePanel';
-import { SessionWrapper } from '../redux';
+import { SessionConfig, SessionWrapper } from '../redux';
 
+const mockConsole = jest.fn(() => null);
 jest.mock('@deephaven/console', () => ({
   ...(jest.requireActual('@deephaven/console') as Record<string, unknown>),
-  Console: jest.fn(() => null),
-  default: jest.fn(() => null),
+  Console: props => mockConsole(props),
+  default: props => mockConsole(props),
 }));
 
-jest.mock('./Panel', () => jest.fn(() => null));
-
-function makeSession() {
-  return {
+function makeSession(): IdeSession {
+  return ({
     addEventListener: jest.fn(),
     subscribeToFieldUpdates: jest.fn(() => () => null),
     removeEventListener: jest.fn(),
     getTable: jest.fn(),
     getObject: jest.fn(),
     runCode: jest.fn(),
-  };
+  } as unknown) as IdeSession;
```

Comment:
Would be better to just mock all of `IdeSession` if it's not a pain. Shouldn't really affect the tests, but if `IdeSession` changes shape we don't know about it in the tests unless they fail

Reply:
It's a bit of a pain in the ass. class Table doesn't match the types, missing many methods, and doesn't see the static events (since our typings expect them to be properties instead of static). Cleaning up mock should happen after we start using the actual types published by JS API.
I did a bit of cleanup, but was going down a bit of a yak shaving exercise.